tuf/client_rework/updater_rework.py

```diff
@@ -558,7 +558,7 @@ def _preorder_depth_first_walk(self, target_filepath) -> Dict:
             # And also decrement number of visited roles.
             number_of_delegations -= 1
             delegations = role_metadata.delegations
-            child_roles = delegations.get("roles", [])
+            child_roles = delegations.roles
```

Comment:
delegations can be none

Reply:
Added one additional commit addressing this.